src/UI/Mail/Main.hs

```diff
@@ -67,25 +68,22 @@ renderAttachmentsList :: AppState -> Widget Name
 renderAttachmentsList s =
     let hasFocus = MailListOfAttachments == focusedViewWidget s
         attachmentsList =
-          L.renderList (renderPart charsets) hasFocus (view (asMailView . mvAttachments) s)
+          L.renderList (\isSel -> renderPart charsets isSel . view headers) hasFocus (view (asMailView . mvAttachments) s)
         charsets = view (asConfig . confCharsets) s
     in attachmentsHeader <=> attachmentsList
 
--- TODO: Both these functions are basically duplicates. Use classes for
--- WireEntity and MIMEMessage to don't repeat our selfs?
--- See #264
-renderPart :: CharsetLookup -> Bool -> WireEntity -> Widget Name
-renderPart charsets selected m =
-  let pType = showContentType $ view (headers . contentType) m
-      pFilename = fromMaybe "--" (preview (headers . contentDisposition . filename charsets) m)
+renderPart :: CharsetLookup -> Bool -> Headers -> Widget Name
+renderPart charsets selected hds =
+  let pType = showContentType $ view contentType hds
+      pFilename = maybe "--" takeFileName (preview (contentDisposition . filename charsets) hds)
       listItemAttr = if selected then listSelectedAttr else listAttr
-      attachmentType = txt (if isAttachment' m then "A" else "I")
+      attachmentType = txt (if isAttachment' hds then "A" else "I")
       widget = hBox
         [ padLeftRight 1 attachmentType
         , padRight Max (txt pFilename)
         , txt pType
         ]
   in withAttr listItemAttr widget
 
-isAttachment' :: WireEntity -> Bool
-isAttachment' = has (headers . contentDisposition . dispositionType . filtered (== Attachment))
+isAttachment' :: Headers -> Bool
```

Comment:
This is currently implemented in purebred-email as:
```Haskell
isAttachment :: MIMEMessage -> Bool
```
but I think we should change it to accept `Headers` as the first argument. Do you concur?

Reply:
In my 'stuff' branch it is already generalised to `isAttachment :: HasHeaders a => a -> Bool`.

`Headers` and `Message s a` have instances of `HasHeaders`.

So... soon :)